### PR TITLE
Alter API and DB table for article-circle relationship

### PR DESCRIPTION
**Pull request by robertu7**
* alter `article_circle` table for access type and paywall secret;
* add `Article.access` fields;
* add indexes to `transaction`;

### CODE REVIEW

**Review by jeremyOk — APPROVED** [review]

**Review by guoliu — COMMENTED** [review]
Great job 💪

Should we also update `PutCircleArticlesInput` in mutation `putCircleArticles`, so that it includes access types `public` and `paywall`?

File: src/queries/article/access.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      'circle.state': CIRCLE_STATE.active,
+    })
+    .first()
+
+  if (!articleCircle || !articleCircle.circleId) {
+    return
+  }
+
+  return atomService.circleIdLoader.load(articleCircle.circleId)
+}
```
Comment (devformatters) [inline src/queries/article/access.ts:88]: we don't break resolver down into scripts anymore?

**Review by devformatters — APPROVED** [review]
Cool. 🫁

**Comment by robertu7** [comment]
> Great job 💪
> 
> Should we also update `PutCircleArticlesInput` in mutation `putCircleArticles`, so that it includes access types `public` and `paywall`?

Please check https://github.com/thematters/matters-server/pull/1903

**Pull request merged**